Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -825,11 +825,11 @@ private extension BrowserViewController {
     // them then iOS will actually first open Safari, which then redirects to the app store. This works but it will
     // leave a 'Back to Safari' button in the status bar, which we do not want.
     func isStoreURL(_ url: URL) -> Bool {
-        if url.scheme == "http" || url.scheme == "https" || url.scheme == "itms-apps" {
-            if url.host == "itunes.apple.com" {
-                return true
-            }
+        guard ["http", "http", "itms-apps", "itms-appss"].contains(url.scheme) else { return false }
+        if url.host == "itunes.apple.com" || url.host == "apps.apple.com" {
```

Comment:
The fix LGTM but unfortunately is not working the `url.host` check is failing and is returning false. The issue is that the url.host is returning` itms-apps` as `url.host`

Reply:
Sorry there was a bug in my logic.
It should have returned true unconditionally for "itms-apps" and "itms-appss", but it did not.
and additionally, add a check for the app store shortened URL host

fix here 58293ea622f5e14c8f763f810bc0e15060f0a766 

fyi. url.host does not return `itms-apps`

Reply:
This will pass the test

<img width="793" alt="스크린샷 2023-09-08 오후 12 13 55" src="https://github.com/mozilla-mobile/firefox-ios/assets/13270453/aee67389-4221-4b08-aa21-7b6cbdcc52b6">